src/python/pants/jvm/jdk_rules.py

```diff
@@ -355,12 +377,16 @@ async def jvm_process(bash: BashBinary, request: JvmProcess) -> Process:
         **request.extra_env,
     }
 
+    jvm_options = []
+    if request.use_nailgun:
+        jvm_options = [*jdk.global_jvm_options, *request.extra_jvm_options]
```

Comment:
Only passing the JVM options when `use_nailgun=True` as the suggestion was to implement the arg templating for the nailgun command runner.

Reply:
Mmm, yea: I hadn't thought through that implication. Seems fine for a first cut: can adjust in a followup.

Can you leave a TODO here explaining why?